### ISSUE TITLE
Support for cubical path types

### ISSUE THREAD

**Issue by NickHu** [isovector/cornelis#67]
Hi, thanks for making a great agda plugin for neovim. I was attempting to use your plugin to follow [this tutorial on Cubical Agda](https://github.com/Smaug123/CubicalTutorial), but I believe that Cornelis doesn't expose all the necessary information.

Here is what the situation looks like in Emacs when doing `<C-c> <C-,>`:
![image](https://user-images.githubusercontent.com/450276/183624385-b6e7c622-e2a5-45b9-9b47-cef3f68d6914.png)

Here is the prompt from Cornelis after requesting a type context:
![image](https://user-images.githubusercontent.com/450276/183624631-cb322285-dc1d-4e9f-b867-c9d0faa3f191.png)

The issue is that Cornelis doesn't tell me what the boundaries are, nor does it show me what the constraints are - both of which are pretty necessary to work with cubical path types.

**Comment by isovector** [comment]
I'm happy to look into this. Can you please:

1. make a minimal file with a single hole that exhibits the problem

2. post the output of running `agda --interaction-json`, and then entering this into the prompt:

```
IOTCM "/path/to/your/file.agda" None Direct (Cmd_load "/path/to/your/file.agda" [])
IOTCM "/path/to/your/file.agda" None Direct (Cmd_goal_type_context Simplified 0 noRange "")
```

(after changing `/path/to/your/file.agda` in all three places)

3. Share both the file and the resulting JSON here :)

**Comment by NickHu** [comment]
Hi, here is the minimal file which exhibits the problem:

```agda
{-# OPTIONS --cubical #-}
open import Agda.Builtin.Cubical.Path

hole : {A : Set} → A ≡ A
hole i = {!!}
```

Here is my output from agda:

```
JSON> IOTCM "/tmp/Cubical/Hole.agda" None Direct (Cmd_load "/tmp/Cubical/Hole.agda" [])
{"kind":"Status","status":{"checked":false,"showImplicitArguments":false,"showIrrelevantArguments":false}}
{"kind":"ClearRunningInfo"}
{"kind":"ClearHighlighting","tokenBased":"NotOnlyTokenBased"}
{"debugLevel":1,"kind":"RunningInfo","message":"Checking Hole (/tmp/Cubical/Hole.agda).\n"}
{"kind":"Status","status":{"checked":false,"showImplicitArguments":false,"showIrrelevantArguments":false}}
{"info":{"errors":[{"message":"Failed to solve the following constraints:\n  A = ?0 (i = Agda.Primitive.Cubical.i1) : Set (blocked on _3)\n  A = ?0 (i = Agda.Primitive.Cubical.i0) : Set (blocked on _3)"}],"invisibleGoals":[],"kind":"AllGoalsWarnings","visibleGoals":[{"constraintObj":{"id":0,"range":[{"end":{"col":14,"line":5,"pos":104},"start":{"col":10,"line":5,"pos":100}}]},"kind":"OfType","type":"Set"}],"warnings":[]},"kind":"DisplayInfo"}
{"interactionPoints":[{"id":0,"range":[{"end":{"col":14,"line":5,"pos":104},"start":{"col":10,"line":5,"pos":100}}]}],"kind":"InteractionPoints"}
JSON> IOTCM "/tmp/Cubical/Hole.agda" None Direct (Cmd_goal_type_context Simplified 0 noRange "")
{"kind":"Status","status":{"checked":false,"showImplicitArguments":false,"showIrrelevantArguments":false}}
{"info":{"goalInfo":{"boundary":["i = Agda.Primitive.Cubical.i0 ⊢ A","i = Agda.Primitive.Cubical.i1 ⊢ A"],"entries":[{"binding":"Set","inScope":false,"originalName":"A","reifiedName":"A"},{"binding":"Agda.Primitive.Cubical.I","inScope":true,"originalName":"i","reifiedName":"i"}],"kind":"GoalType","outputForms":["A = ?0 (i = Agda.Primitive.Cubical.i1) : Set (blocked on _3)","A = ?0 (i = Agda.Primitive.Cubical.i0) : Set (blocked on _3)"],"rewrite":"Simplified","type":"Set","typeAux":{"kind":"GoalOnly"}},"interactionPoint":{"id":0,"range":[{"end":{"col":14,"line":5,"pos":104},"start":{"col":10,"line":5,"pos":100}}]},"kind":"GoalSpecific"},"kind":"DisplayInfo"}
```

As you can see from the JSON of the second message, it emits boundary conditions and additional constraints in `goalInfo` but this is not shown by Cornelis.

Comparison of Emacs:
![image](https://user-images.githubusercontent.com/450276/184168951-e7c55401-ed2c-4789-9512-40bffbb92d99.png)

and Neovim + Cornelis:
![image](https://user-images.githubusercontent.com/450276/184168827-ad0483cb-803e-49d3-af68-48936e137506.png)

**Comment by isovector** [comment]
Should be fixed in b2a6adeec6097d99c6a99f86283b145525bc1367. Mind giving it a play?

New output:

```
Boundary:
i = Agda.Primitive.Cubical.i0 ⊢ A
i = Agda.Primitive.Cubical.i1 ⊢ A
————————————————————————————————————————————————————————————
Goal: Set
————————————————————————————————————————————————————————————
(A) : Set
i : Agda.Primitive.Cubical.I
————————————————————————————————————————————————————————————
Constraints:
A = ?0 (i = Agda.Primitive.Cubical.i1) : Set
  (blocked on _3@1490201231961467499)
A = ?0 (i = Agda.Primitive.Cubical.i0) : Set
  (blocked on _3@1490201231961467499)

```

**Comment by NickHu** [comment]
It's looking great, thanks! I'm not an expert on cubical agda, but from my perspective it looks really usable now. Thanks a lot for the lightning-fast response!
